sparse/coo/core.py

```diff
@@ -504,15 +507,17 @@ def __getitem__(self, index):
 
         last_ellipsis = len(index) > 0 and index[-1] is Ellipsis
         index = normalize_index(index, self.shape)
-        if len(index) != 0 and all(not isinstance(ind, Iterable) and ind == slice(None) for ind in index):
+
+        if len(index) != 0 and all(not isinstance(ind, Iterable) and ind == slice(0, dim, 1)
+                                   for ind, dim in zip_longest(index, self.shape)):
             return self
-        mask = np.ones(self.nnz, dtype=np.bool)
-        for i, ind in enumerate([i for i in index if i is not None]):
-            if not isinstance(ind, Iterable) and ind == slice(None):
-                continue
-            mask &= _mask(self.coords[i], ind, self.shape[i])
+        mask = _mask(self.coords, index, self.shape)
+
+        if isinstance(mask, slice):
+            n = len(range(mask.start, mask.stop, mask.step))
```

Comment:
This may be somewhat inefficient in Python 2

Reply:
In Python 2, we are using `xrange` (imported from `compatibility.py` as `range`). It doesn't construct a list here, if that's what you're worried about.